### PR TITLE
Add ability to lock post comments to new accounts

### PR DESCRIPTION
**Pull request by Will-Howard**
for this issue: https://github.com/ForumMagnum/ForumMagnum/issues/5344

The following features already existed:

* the ability to lock comments on a post completely
* the ability to lock replies to a comment completely until a certain date

so I have just added the ability to block replies on a post to accounts created after a certain date (any date, not necessarily the createdAt date of the post). I could also add the same thing to individual comments fairly easily if anyone thinks that would be useful

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1202725388297843) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by Raemon** [comment]
I think there might already be a feature doing this, or doing something pretty similar. I'll look into it soon to see if I'm remembering right.

File: packages/lesswrong/lib/collections/users/helpers.tsx [hunk 1]
```diff
@@ -184,27 +186,32 @@ export const userIsAllowedToComment = (user: UsersCurrent|DbUser|null, post: Pos
   return true
 }
 
-export const userBlockedCommentingReason = (user: UsersCurrent|DbUser|null, post: PostsDetails|DbPost, postAuthor: PostsAuthors_user|null): string => {
+export const userBlockedCommentingReason = (user: UsersCurrent|DbUser|null, post: PostsDetails|DbPost, postAuthor: PostsAuthors_user|null): JSX.Element => {
```
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/users/helpers.tsx:189]: You might consider having the return type be string|JSX.Element, but this is fine too.

File: packages/lesswrong/lib/collections/posts/custom_fields.ts [hunk 2]
```diff
@@ -682,6 +682,15 @@ addFieldsDict(Posts, {
     optional: true,
     control: "checkbox",
   },
+  commentsLockedToAccountsCreatedAfter: {
+    type: Date,
+    control: 'datetime',
+    viewableBy: ['guests'],
+    group: formGroups.moderationGroup,
+    insertableBy: (currentUser: DbUser|null) => userCanCommentLock(currentUser, null),
```
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/posts/custom_fields.ts:690]: Did not remember that group existed. Nice work.

**Review by jpaddison3 — APPROVED** [review]
Perfect job, nice work

**Pull request merged**